awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -438,6 +446,15 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
                 self.json_output['id'] = response['json']['id']
                 self.json_output['changed'] = True
                 item_url = response['json']['url']
+            # 200 is response from approval node creation
+            elif response['status_code'] == 200 and item_type == 'workflow_job_template_approval_node':
+                self.json_output['name'] = 'unknown'
+                for key in ('name', 'username', 'identifier', 'hostname'):
+                    if key in response['json']:
+                        self.json_output['name'] = response['json'][key]
+                self.json_output['id'] = response['json']['id']
+                self.json_output['changed'] = True
+                item_url = response['json']['url']
             else:
```

Comment:
Can we handle this in the code after the on_continue? This seems very specific to workflow_job_template_approval_nodes and I don't think we want to fill the common library with specific one off behaviors.

Reply:
We could definitly put the logic after on continue, but the item_url still needs population and it will be different for the workflow approval, and its a check if it came back ok before attempting associations. This I'm not so sure on moving, as it logically makes sense to keep the block of 441-464 together.

Reply:
I guess I am also curious why this endpoint is returning a 200 instead of a 201. That is probably more of a Tower question than a collection question but it's breaking our "pattern" of expecting a 201 to be returned for something new. @ryanpetrello any idea why this endpoint would return a 200 instead of a 201 for an on create? Any idea if there are other endpoint that return a 200 instead of a 201?

We have, on other functions, done things like `if response['status_code'] in [200, 201]:` if the return code varies. Or, if we really need to be more specific for this use case could we do something like:
```
if response['status_code'] == 201  or (response['status_code'] == 200 and item_type == 'workflow_job_template_approval_node'):
```

That would eliminate redundant code (let me know if I am misreading and if and elif are not identical code).

Reply:
Generally speaking, our endpoints that *create* things should be returning HTTP 200.  HTTP 201 also isn't uncommon, but it's generally reserved for situations where we've created something that's still pending, like job launches:

```
awx/api/generics.py
626:        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
664:            if response.status_code != status.HTTP_201_CREATED:
698:            return Response(data, status=status.HTTP_201_CREATED, headers=headers)
1024:        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)

awx/api/views/webhooks.py
47:        return Response({'webhook_key': obj.webhook_key}, status=status.HTTP_201_CREATED)

awx/api/views/__init__.py
2431:            return Response(data, status=status.HTTP_201_CREATED, headers=headers)
2887:        return Response(status=status.HTTP_201_CREATED, headers=headers)
3222:        return Response(data, status=status.HTTP_201_CREATED, headers=headers)
3254:        return Response(data, status=status.HTTP_201_CREATED, headers=headers)
3452:        return Response(data, status=status.HTTP_201_CREATED, headers=headers)
3640:            return Response(data, status=status.HTTP_201_CREATED, headers=headers)
3698:        return Response(data, status=status.HTTP_201_CREATED, headers=headers)
3878:        if response.status_code != status.HTTP_201_CREATED:
3985:            return Response(data, status=status.HTTP_201_CREATED, headers=headers)
```

Reply:
Hrm...

Reply:
This is probably why:

![image](https://user-images.githubusercontent.com/214912/96505738-900fa880-1224-11eb-8097-376c635ac2e6.png)

Should we just change this API code to be a 201?

Reply:
Here ya go, let's do this instead: https://github.com/ansible/awx/pull/8416

Reply:
I think that would work, I was wondering it myself @john-westcott-iv the 200+item was more being overly cautious, @ryanpetrello working on new build with John's suggestions where I change this to 201